### PR TITLE
Adds error checking to Form 8. closes #174

### LINKED ISSUE #174: Add JavaScript form error handling
Build on HTML5 APIs.

- [x] Make linked text fields required if shown.
- [x] Of course, that means adding a method to add / remove hidden attribute

Refer to messaging and indicators from #231, #230.

### PR DESCRIPTION
**Pull request by webinista**
Also adds CSS for radio button errors.

screen shot 2016-01-11 at 16 32 37

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -269,3 +288,69 @@ $(function() {
 $(document).on('ready', function(e){
     $('.cf-form-cond-req').trigger('input');
 })
+
+
+/*----------------------------------
+ * Error handling for questions form
+ * ----------------------------------*/
+$(function(){
+    var ERROR_MESSAGES = {
```
Comment (plusjeff) [inline app/assets/javascripts/application.js]: For the `ERROR_MESSAGES`, can the `<div>`s be moved into the templates, and have IDs attributes added such that they can be shown/hidden? This helps with the separation of concerns on the page (view vs behavior).

**Comment by plusjeff** [comment]
Added a comment, but assign back to me response and/or fix.

File: app/assets/javascripts/application.js (hunk 1 above)
Comment (webinista) [inline app/assets/javascripts/application.js]: I moved the containing `<div>s` into the view, which allowed for some JS cleanup. Did you also mean that the text itself should be moved into a view?

Comment (plusjeff) [inline app/assets/javascripts/application.js]: That's probably good for now. Maybe in the future, we could move that into a i18n-ish kind of file (or some kind of key/value space for just view content).

**Comment by plusjeff** [comment]
Going to bounce over to @shanear for review.

**Pull request merged**